### PR TITLE
Add aten::argmin, aten::aminmax, aten::frac

### PR DESCRIPTION
**Pull request by hjhee**
- _aminmax
  - _aminmax.dim
  - aminmax
  - aminmax.out
  - argmin
  - argmin.out
  - frac
  - frac_
  - frac.out

### CODE REVIEW

File: src/ATen/native/xpu/TensorCompare.cpp [hunk 1]
```diff
@@ -568,4 +568,11 @@ ::std::tuple<Tensor&, Tensor&> XPUNativeFunctions::max_out(
   return {values, indices};
 }
 
+std::tuple<Tensor, Tensor> XPUNativeFunctions::_aminmax(
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim) {
```
Comment (xytintel) [inline src/ATen/native/xpu/TensorCompare.cpp:574]: Pls add warning

File: src/ATen/native/xpu/ReduceOps.cpp [hunk 2]
```diff
@@ -803,4 +832,80 @@ Tensor XPUNativeFunctions::amin(
   return out;
 }
 
+TensorIterator meta_aminmax(
```
Comment (xytintel) [inline src/ATen/native/xpu/ReduceOps.cpp:835]: Pls follow the latest aminmax meta func

File: src/ATen/native/xpu/ReduceOps.cpp [hunk 3]
```diff
@@ -803,4 +832,80 @@ Tensor XPUNativeFunctions::amin(
...
+}
+
+std::tuple<Tensor&, Tensor&> XPUNativeFunctions::aminmax_out(
+    const Tensor& self,
+    std::optional<int64_t> dim_opt,
+    bool keepdim,
+    Tensor& min,
+    Tensor& max) {
+  auto iter = meta_aminmax(self, dim_opt, keepdim, min, max);
+  native::xpu::aminmax_kernel(iter);
```
Comment (xytintel) [inline src/ATen/native/xpu/ReduceOps.cpp:907]: Pls follow the latest pytorch's impl

File: src/ATen/native/xpu/UnaryOps.cpp [hunk 4]
```diff
@@ -627,7 +649,6 @@ Tensor& XPUNativeFunctions::cosh_(Tensor& self) {
 Tensor& XPUNativeFunctions::cosh_out(const Tensor& self, Tensor& out) {
   TensorIterator iter;
   iter.build_borrowing_unary_float_op(out, self);
```
Comment (xytintel) [inline src/ATen/native/xpu/UnaryOps.cpp:651]: Where is the code?

File: src/ATen/native/xpu/sycl/ReduceAMinMaxKernel.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#include <ATen/NumericUtils.h>
+#include <ATen/native/SharedReduceOps.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/xpu/sycl/NumericLimits.h>
+#include <ATen/native/xpu/sycl/Reduce.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+void aminmax_kernel_impl(TensorIterator& iter) {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/ReduceAMinMaxKernel.cpp:11]: Pls align to cuda

File: src/ATen/native/xpu/sycl/ReduceAMinMaxKernel.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+void aminmax_kernel_impl<bool>(TensorIterator& iter) {
+  gpu_reduce_kernel<uint8_t, uint8_t>(
+      iter,
+      MinMaxOps<uint8_t, uint8_t, int32_t>{},
+      std::pair<uint8_t, uint8_t>(
+          at::numeric_limits<uint8_t>::upper_bound(),
+          at::numeric_limits<uint8_t>::lower_bound()));
+}
+
+void aminmax_kernel(TensorIterator& iter) {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/ReduceAMinMaxKernel.cpp:31]: Pls align to cuda

**Review by xytintel — APPROVED** [review]

**Comment by fengyuan14** [comment]
Cannot reproduce the failure in preci at local,
image

**Pull request merged**